### PR TITLE
New options for GMCJDriver

### PR DESCRIPTION
**Pull request by pochoarus**
New options:
-force the interaction.
-use logarithmic binning in pmax.
-change the safety factor.

### CODE REVIEW

File: src/Framework/EventGen/GMCJDriver.cxx [hunk 1]
```diff
@@ -641,13 +679,23 @@ void GMCJDriver::ComputeProbScales(void)
...
+  }
+  else {
+    // for maximum interaction probability vs E /for given geometry/ I will
+    // be using 300 bins up to the maximum energy for the input flux
+    double de   = fEmax/300.;//djk june 5, 2013
+    double emin = 0.0;
+    double emax = fEmax + de;
+    n = 1 + (int) ((emax-emin)/de);
+    for(int i=0; i<n; i++) ebins[i] = emin + i * (emax-emin)/n;
+  }
```
Comment (nusense) [inline src/Framework/EventGen/GMCJDriver.cxx:698]: I know `300` is in the original code, but could this also be another parameter (with the default 300) and this size should be used to allocate the `ebins` array.  But first the filling of `ebins` goes one beyond `300` .e.g.  `(int i=0; i<=n; i++)`  which the saving grace is that `bins` is allocated to be `[400]` (could this be `[n+1]` if `n=300`?).  I'm presuming that `n=`+(int)((emax-emin)/de) is going to give `301`

Also verify that `ebins` is deleted somewhere later (i.e. not a memory leak)

File: src/Framework/EventGen/GMCJDriver.cxx [hunk 2]
```diff
@@ -125,6 +125,36 @@ void GMCJDriver::KeepOnThrowingFluxNeutrinos(bool keep_on)
...
+//___________________________________________________________________________
+void GMCJDriver::ForceInteraction()
+{
+// Use a single probability scale. That generates unweighted events.
+// (Note that generating unweighted event kinematics is a different thing)
+//
+  fForceInteraction = true;
+
+  LOG("GMCJDriver", pNOTICE)
+    << "GMCJDriver will generate weighted events forcing the interaction. ";
```
[the remark this answers is not in the document]
Comment (pochoarus) [reply src/Framework/EventGen/GMCJDriver.cxx]: fixed

File: src/Framework/EventGen/GMCJDriver.cxx [hunk 3]
```diff
@@ -605,7 +639,11 @@ void GMCJDriver::BootstrapXSecSplineSummation(void)
     double dE  = fEmax/10.;
     double min = rE.min;
     double max = (fEmax+dE < rE.max) ? fEmax+dE : rE.max;
-    evgdriver->CreateXSecSumSpline(100,min,max,true);
+    
+    // in the logaritmic binning is important to have a narrow binning to
+    // describe better the glashow resonance peak.
+    int nbins = (fPmaxLogBinning) ? 1000 : 100;
```
[the remark this answers is not in the document]
Comment (pochoarus) [reply src/Framework/EventGen/GMCJDriver.cxx]: i've created a new parameter fXSecSplineNbins for that purpose

File: src/Framework/EventGen/GMCJDriver.cxx (hunk 1 above)
Comment (pochoarus) [reply src/Framework/EventGen/GMCJDriver.cxx]: fixed

**Comment by pochoarus** [comment]
i've modified the code in order to address robert's suggestion.

**Pull request merged**